pydantic/_internal/_docs_extraction.py

```diff
@@ -0,0 +1,79 @@
+"""Utilities related to attribute docstring extraction."""
+from __future__ import annotations
+
+import ast
+import inspect
+import re
+import textwrap
+from typing import Any
+
+
+class DocstringVisitor(ast.NodeVisitor):
+    def __init__(self) -> None:
+        super().__init__()
+
+        self.target: str | None = None
+        self.attrs: dict[str, str] = {}
+        self.previous_node_type: type[ast.AST] | None = None
+
+    def visit(self, node: ast.AST) -> Any:
+        node_result = super().visit(node)
+        self.previous_node_type = type(node)
+        return node_result
+
+    def visit_AnnAssign(self, node: ast.AnnAssign) -> Any:
+        if isinstance(node.target, ast.Name):
+            self.target = node.target.id
+
+    def visit_Expr(self, node: ast.Expr) -> Any:
+        if isinstance(node.value, ast.Str) and self.previous_node_type is ast.AnnAssign:
+            docstring = inspect.cleandoc(node.value.s)
+            if self.target:
+                self.attrs[self.target] = docstring
+            self.target = None
+
+
+def _extract_source_from_frame(cls_name: str) -> list[str] | None:
+    frame = inspect.currentframe()
+
+    while frame:
+        lnum = frame.f_lineno
+        try:
+            lines, _ = inspect.findsource(frame)
+        except OSError:
+            # Source can't be parsed (maybe because running in an interactive terminal),
+            # we don't want to error here.
+            continue
```

Comment:
I think the tests are currently stuck in an infinite loop at this point.

Are you able to run tests locally before you push?